File: vol1/ch7/LeetCode538.py
class Solution:
    def convertBST(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        self.sum=0
        self.revinorder(root)
        return root
    
    def revinorder(self,root):
        if root!=None:
            self.revinorder(root.right)
            self.sum+=root.val
            root.val=self.sum
            self.revinorder(root.left)

File: vol1/ch7/LeetCode538.cpp
class Solution {
    int sum;
public:
    TreeNode* convertBST(TreeNode* root) {
        sum=0;
        revinorder(root);
        return root;
    }

    void revinorder(TreeNode* root) {
        if(root!=NULL) {
	        revinorder(root->right);
     	   sum+=root->val;
        		root->val=sum;
        	revinorder(root->left);
		}
    }
};

